packages/network-controller/src/NetworkController.test.ts

```diff
@@ -1,224 +1,3653 @@
+
+type WithControllerArgs<ReturnValue> =
+  | [WithControllerCallback<ReturnValue>]
+  | [WithControllerOptions, WithControllerCallback<ReturnValue>];
 
+// Store this in case it gets stubbed later
+const originalSetTimeout = global.setTimeout;
+const SubproviderMock = mocked(Subprovider);
+const createInfuraProviderMock = mocked(createInfuraProvider);
+const createMetamaskProviderMock = mocked(createMetamaskProvider);
+
+//                                                                                     setProviderType            setRpcTarget
+//                                                                                            └───────────┬────────────┘
+// set providerConfig                                                                               refreshNetwork
+//       │ │ └────────────────────────────────────────────┬──────────────────────────────────────────────┘ │
+//       │ │                                     initializeProvider                                        │
+//       │ │                  ┌─────────────────────────┘ │ └─────────────────────────┐                    │
+//       │ │          setupInfuraProvider        setupStandardProvider      getEIP1559Compatibility        │
+//       │ │                  └─────────────┬─────────────┘                                                │
+//       │ │                          updateProvider                                                       │
+//       │ └───────────────┬───────────────┘ └───────────────────────────────┐                             │
+//       │          registerProvider                                  this.provider = ...                  │
+//       │                 ⋮                                                                               │
+//       │   this.provider.on('error', ...)                                                                │
+//       │                 │                                                                               │
+//       │            verifyNetwork                                                                        │
+//       │                 └─────────────────────────────┐                                                 │
+//       └───────────────────────────────────────────────┼─────────────────────────────────────────────────┘
+//                                                 lookupNetwork
+
+describe('NetworkController', () => {
   beforeEach(() => {
-    messenger = new ControllerMessenger().getRestricted({
-      name: 'NetworkController',
-      allowedEvents: ['NetworkController:providerConfigChange'],
-      allowedActions: [],
-    });
+    jest.resetAllMocks();
+    jest.useFakeTimers();
   });
 
   afterEach(() => {
-    sinon.restore();
+    jest.useRealTimers();
   });
 
-  it('should set default state', () => {
-    const controller = new NetworkController({
-      messenger,
-      infuraProjectId: 'potate',
+  describe('constructor', () => {
```

Comment:
Nit: It seems that the actions `getProviderConfig` and `getEthQuery` aren't covered by any tests

Reply:
The `infuraProjectId` parameter isn't covered by these constructor tests either. Though it is covered by the tests for other methods at least.

Reply:
Alas... I didn't have time to take care of these items today. I'll have to finish this next week.

Reply:
Tests added here: [1d26cea](https://github.com/MetaMask/core/pull/1039/commits/1d26cea0ffb64f5e592dd21de6c45a07d5ab6502)